### PR TITLE
Hamiltonian simulation guide

### PR DESCRIPTION
**Pull request by alexandre-ricardo**
Adding a first draft for the Hamiltonian Simulation guide. Not yet finished.

### CODE REVIEW

File: tutorials/hamiltonian_simulation/hamiltonian_simulation_guide.ipynb [hunk 1]
```diff
@@ -0,0 +1,604 @@
+{
```
Comment (TomerGoldfriend) [inline tutorials/hamiltonian_simulation/hamiltonian_simulation_guide.ipynb:1]: <p>Maybe mention that given a Suzuki-Trotter formula, there are exponential different ways to implement it. The synthesis engine generates an optimized implementation, reducing CX-counts and depth compared to a naive, non-optimized implementation.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/201/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (TomerGoldfriend) [inline tutorials/hamiltonian_simulation/hamiltonian_simulation_guide.ipynb:1]: <p>and apply it ---&gt; and applies it</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/201/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (TomerGoldfriend) [inline tutorials/hamiltonian_simulation/hamiltonian_simulation_guide.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #10.            num_qdrift=30,
</pre><p>why did you choose 30? is it according to the error formula you have written above?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/201/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (TomerGoldfriend) [inline tutorials/hamiltonian_simulation/hamiltonian_simulation_guide.ipynb:1]: <p>Nice plot. I would increase <code>num_qdrift</code> , or, alternatively, use <code>num_qdrift=30</code> as you did but run the simulation several times to get some statistics (this is eventually a random approach). You can use <code>ExecutionSession</code> .  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/201/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

Comment (TomerGoldfriend) [inline tutorials/hamiltonian_simulation/hamiltonian_simulation_guide.ipynb:1]: <p>The notebook in the link shows two things:</p><ol><li>How to use these methods for Hamiltonian derived from molecules.</li><li>The optimization of these functions by the synthesis engine.</li></ol><p>I am not sure it gives more experience, it mainly highlights the latter (which is a good thing to mention).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Classiq/classiq-library/pull/201/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

**Review by EdenSchirmanClassiq — APPROVED** [review]
Great

**Pull request merged**